src/kayenta/metricStore/signalfx/metricConfigurer.tsx

```diff
@@ -0,0 +1,204 @@
+const getQueryPairs = (metric: ICanaryMetricConfig) => get(metric, 'query.queryPairs', []) as IKeyValuePair[];
+const getAggregationMethod = (metric: ICanaryMetricConfig) => get(metric, 'query.aggregationMethod', '');
+
+export const SIGNAL_FLOW_SIMPLE = 'signalfx';
+
+@autoBindMethods
+class SignalFxMetricConfigurer extends React.Component<SignalFxMetricConfigurerProps> {
+
+  public onMetricNameChange(e: React.ChangeEvent<HTMLInputElement>) {
+    this.props.updateMetricName(e.target.value);
+  }
+
+  public onAggregationMethodChange(e: React.ChangeEvent<HTMLInputElement>) {
+    this.props.updateAggregationMethod(e.target.value);
+  }
+
+  public render() {
+
+    const { editingMetric, updateQueryPairs, validationErrors } = this.props;
+
+    return (
+      <section>
+        <FormRow
+          label="SignalFx Metric"
+          error={get(validationErrors, 'signalFxMetric.message', null)}
+        >
+          <DisableableInput
+            type="text"
+            value={getSignalFxMetric(editingMetric)}
+            onChange={this.onMetricNameChange}
+            disabledStateKeys={[DISABLE_EDIT_CONFIG]}
+          />
+        </FormRow>
+        <FormRow
+          label={<>Aggregation Method <HelpField id="canary.config.signalFx.aggregationMethod"/></>}
+          error={get(validationErrors, 'aggregationMethod.message', null)}
+        >
+          <DisableableInput
+            type="text"
+            value={getAggregationMethod(editingMetric)}
+            onChange={this.onAggregationMethodChange}
+            disabledStateKeys={[DISABLE_EDIT_CONFIG]}
+          />
+        </FormRow>
+        <FormRow
+          label={<>Query Pairs <HelpField id="canary.config.signalFx.queryPairs"/></>}
+          error={get(validationErrors, 'queryPairs.message', null)}
+        >
+          <KeyValueList className="signalfx-query-pairs" list={getQueryPairs(editingMetric)} actionCreator={updateQueryPairs}/>
+        </FormRow>
+      </section>
+    );
+  }
+}
+
+/**
+ * Validates SignalFx specific fields on the edit metric modal
+ */
+export function validateMetric(editingMetric: ICanaryMetricConfig): ICanaryMetricValidationErrors {
+  const errors: ICanaryMetricValidationErrors = {};
+
+  const validators: MetricValidatorFunction[] = [
+    ...getSignalFxValidators(editingMetric)
+  ];
+
+  return validators.reduce(
+    (reducedErrors, validator) => validator(reducedErrors, editingMetric),
+    errors,
+  );
+}
+
+
+/**
+ * Get validators based on type
+ */
+export function getSignalFxValidators(editingMetric: ICanaryMetricConfig): MetricValidatorFunction[] {
+  if (!editingMetric || !editingMetric.query) {
+    return [];
+  }
+  const { type } = editingMetric.query;
+  return signalFxMetricValidators[type] || [];
+}
+
+/**
+ * Collection of the validators the SignalFx metricConfigurer cares about.
+ */
+export const signalFxMetricValidators: { [key: string]: MetricValidatorFunction[] } = {
+  [SIGNAL_FLOW_SIMPLE]: [
```

Comment:
This is a little confusing to me. Are there other signal fx types that are going to be using this component (but not using these validators)?

Reply:
Up to you. If you leave it, could you please leave a comment describing what you're planning?

Reply:
I removed the unneeded logic for now.